File: 1367-linked-list-in-binary-tree/1367-linked-list-in-binary-tree.cpp
class Solution {
    bool chk(ListNode* listNode, TreeNode* treeNode) {
        if (listNode == NULL) return true;
        if (treeNode == NULL || listNode->val != treeNode->val) return false;
        return chk(listNode->next, treeNode->left) || chk(listNode->next, treeNode->right);
    }

public:
    bool isSubPath(ListNode* head, TreeNode* root) {
        if (root == NULL) return false;
        if (chk(head, root)) return true;
        return isSubPath(head, root->left) || isSubPath(head, root->right);
    }
};

File: 1367-linked-list-in-binary-tree/1367-linked-list-in-binary-tree.py
class Solution:
    def isSubPath(self, head: Optional[ListNode], root: Optional[TreeNode]) -> bool:
        def chk(listNode, treeNode):
            if not listNode: return True
            if not treeNode or listNode.val != treeNode.val: return False
            return chk(listNode.next, treeNode.left) or chk(listNode.next, treeNode.right)

        if not root: return False
        if chk(head, root): return True
        return self.isSubPath(head, root.left) or self.isSubPath(head, root.right)
